### PR TITLE
[Usage] Collect failure message

### PR DESCRIPTION
**Pull request by Michaelvll**
We used `sys.exit(1)` for CLIs like `sky local up` which will cause the error message failed to be exposed by the usage collection. We now change them to error raising for more complete error message.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
  - [x] Manually added an error in `local up` script and check the failure appear in usage collection
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: tests/test_jobs_and_serve.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def test_cancel_on_jobs_controller(self, _mock_cluster_state,
                                        _mock_jobs_controller):
         cli_runner = cli_testing.CliRunner()
         result = cli_runner.invoke(cli.cancel, [jobs.JOB_CONTROLLER_NAME, '-a'])
-        assert result.exit_code == 1
+        assert result.exit_code == 2
```
Comment (romilbhardwaj) [inline tests/test_jobs_and_serve.py:257]: Is this change intended?

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll!

File: tests/test_jobs_and_serve.py (hunk 1 above)
Comment (Michaelvll) [reply tests/test_jobs_and_serve.py]: Good catch! Changed it to `click.UsageError.exit_code` for better readability
